sopel/coretasks.py

```diff
@@ -832,6 +850,46 @@ def sasl_success(bot, trigger):
     bot.write(('CAP', 'END'))
 
 
+@module.event(events.RPL_SASLMECHS)
+def sasl_mechs(bot, trigger):
+    # Presumably we're only here if we said we actually *want* sasl, but still
+    # check anyway in case the server glitched.
+    password = None
+    mech = None
+    if bot.config.core.auth_method == 'sasl':
+        password = bot.config.core.auth_password
+        mech = bot.config.core.auth_target
+    elif bot.config.core.server_auth_method == 'sasl':
+        password = bot.config.core.server_auth_password
+        mech = bot.config.core.server_auth_sasl_mech
+    if not password:
+        return
+
+    supported_mechs = trigger.args[1].split(',')
+    if mech not in supported_mechs:
+        """
+        How we get here:
+
+        1. Sopel connects to a network advertising SASL 3.1
+        2. SASL 3.1 doesn't advertise supported mechanisms up front, so Sopel
+           blindly goes ahead with whatever SASL config it's set to use
+        3. The server doesn't support the mechanism Sopel used, and is a good
+           IRC citizen, so it sends this optional numeric, 908 RPL_SASLMECHS
+
+        Note that misconfigured SASL 3.1 will just silently fail when connected
+        to an IRC server NOT implementing the optional 908 reply.
+        A network with SASL 3.2 should theoretically never get this far because
+
```

Comment:
```suggestion
```

Reply:
I've no idea how that happened 👀; will fix.

Though now I've noticed this seven-year-old extra blank in `blocks()`: https://github.com/sopel-irc/sopel/blob/b16cfcd8e1197e4e85727f00f56c69f01c890aac/sopel/coretasks.py#L843-L848